cmd/cli/serve/util.go

```diff
@@ -167,16 +170,46 @@ func getAllowListedLocalPathsConfig() []string {
 	return viper.GetStringSlice(types.NodeAllowListedLocalPaths)
 }
 
-func getNetworkConfig() (node.NetworkConfig, error) {
+func getNetworkConfig(nodeID string) (node.NetworkConfig, error) {
 	var networkCfg types.NetworkConfig
 	if err := config.ForKey(types.NodeNetwork, &networkCfg); err != nil {
 		return node.NetworkConfig{}, err
 	}
+
+	if networkCfg.AuthSecret == "" {
```

Comment:
What's your thinking on that?

AFAICS it's simpler to assume the connection will always be secure, and it doesn't add any extra burden on the user – the token is wrapped up in the orchestrator URL issued by the requester on the terminal. 

From my POV, this is the same as the behaviour we discussed for API TLS – the server generates a cert from its key, and the gives the client a fingerprint which it can use to connect and verify the connection.

Reply:
> the token is wrapped up in the orchestrator URL issued by the requester on the terminal.

I hadn't realized this. lgtm. (seeing the comments for it now in the config file)